Add additional triggers for ServiceHistoryEnrollment processing

gigxz opened the pull request:
## _Merging this PR_
- use the squash-merge strategy

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
Issue: https://github.com/open-path/Green-River/issues/6884

* Add a few missing fields to `warehouse_columns_changed?` methods. These are fields that are referenced by ServiceHistoryEnrollment rebuild task.
* This fixes bug whereby changing certain fields in HMIS, like Move-in Date, would not be propagated to ServiceHistoryEnrollments and thereby would not be displayed in reporting.
* Also add comments and tests

### How to test
- change move-in date
- see if SHE build is queued
- when SHE build is done processing, the local SHE record should have updated move-in date value. (find by "enrollment_group_id"==Enrollment.EnrollmentID)

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
Bug fix

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [na] If adding a new endpoint / exposing data in a new way, I have:
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

gigxz committed:
Add additional triggers for ServiceHistoryEnrollment processing

gigxz committed:
Add tests

gigxz committed:
Update wip case

ttoomey reviewed (approved):
Looks good, thanks for adding tests!

eanders reviewed (approved):
Thanks @gigxz this looks like what we need.  
Two other scenarios we should confirm are covered:
1. Any change to the makeup of a household should trigger a rebuild
2. If a project moves between organizations, we should invalidate all enrollments.  (I think the project cleanup script will catch this, but no harm in queuing the work when it happens.)

gigxz commented:
Thanks Elliot! Notes below and I will copy them to the issue:

> 1. Any change to the makeup of a household should trigger a rebuild

I think we have the triggers covered for all the below scenarios (except partially # 4). This is assuming that we will fix the issue with https://github.com/greenriver/hmis-warehouse/pull/4815#discussion_r1816636991 , so that the invalidation of one Enrollment will in fact invalidate and re-process the whole household.

1. A new enrollment is created in the household and "auto entered" (processing trigger: new Enrollment creation)
2. A WIP enrollment in the household is fully entered (processing trigger: `ProjectID`)
3. An Enrollment in the household is deleted (processing trigger: `DateDeleted`)
4. An Enrollment is "split out" from the household to another new or existing household (processing trigger: `HouseholdID` for the enrollment that's split out, **but** we need another step to ensure that the "source" household is rebuilt as well. I added a note to https://github.com/open-path/Green-River/issues/5800 about that)
5. Two households are "merged" (processing trigger: `HouseholdID`)
6. Relationship to HoH changes on any household member (processing trigger: `RelationshipToHoH`)



> 2. If a project moves between organizations, we should invalidate all enrollments. (I think the project cleanup script will catch this, but no harm in queuing the work when it happens.)

Added a note to https://github.com/open-path/Green-River/issues/5767 and https://github.com/open-path/Green-River/issues/6246
